File: 345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vow = "aeiouAEIOU"

        s = list(s)
        l, r = 0, len(s) - 1

        while l < r:
            if s[l] in vow:
                if s[r] in vow:
                    s[l], s[r] = s[r], s[l]
                    l += 1 
                    r -= 1
                else:
                    r -= 1
            else:
                l += 1
            
        return "".join(s)

File: 345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        int l =0, r = s.length() -1;
        char[] alpha = s.toCharArray();

        while(l<r){
            while(l<r && ! isVowel(alpha[l])){
                l++;
            }
            while(l<r && ! isVowel(alpha[r])){
                r--;
            }
            char temp = alpha[l];
            alpha[l] = alpha[r];
            alpha[r] = temp;
            l++;
            r--;
        }
        return String.valueOf(alpha);
    }

    public boolean isVowel(char ch){
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U';
    }

}